google-cloud-clients/google-cloud-spanner/src/main/java/com/google/cloud/spanner/ReadContext.java

```diff
@@ -113,6 +162,20 @@ ResultSet readUsingIndex(
    * com.google.spanner.v1.ResultSetStats} that can be accessed by calling {@link
    * ResultSet#getStats()} on the returned {@code ResultSet}.
    *
+   * <!--SNIPPET read_context_analyze_query-->
+   * <pre>{@code
+   * ReadContext rc = dbClient.singleUse();
+   * ResultSet resultSet =
+   *     rc.analyzeQuery(
+   *         Statement.of("SELECT SingerId, AlbumId, MarketingBudget FROM Albums"),
+   *         ReadContext.QueryAnalyzeMode.PROFILE);
+   * while (resultSet.next()) {
+   *   resultSet.getCurrentRowAsStruct();
```

Comment:
you may add a comment here as to why discard all the intermediate resutl rows because the Stats are only provided in the last index